numpy/_core/include/numpy/utils.h

```diff
@@ -34,4 +36,15 @@
 #define NPY_CAT_(a, b) NPY_CAT__(a, b)
 #define NPY_CAT(a, b) NPY_CAT_(a, b)
 
+static char get_installed_numpy_major_version()
+{
+    PyObject *numpy = PyImport_ImportModule("numpy");
+    PyObject *version_obj = PyObject_GetAttrString(numpy, "__version__");
+    Py_DECREF(numpy);
+    char major_version = PyUnicode_ReadChar(version_obj, 0);
+    Py_DECREF(version_obj);
+
+    return major_version;
+}
```

Comment:
Using this to get the major version is a bit overloaded IMO, since we also have it available as a C function in the API table.
So, I think this helper should just import `_multiarray_umath`, that will also simplify all other code anyway and separate the concern of what the major version is.
Maybe `PyObject_HasAttr(numpy, "_core")` is enough for a check?

That said, we can probably probably add such a helper, although I think it should also be called `_npy_...` to more clearly signal that it is off-limits and a NumPy implementation detail.

Reply:
Ah, using `PyObject_HasAttr(numpy, "_core")` looks way simpler and it does the same job.

Reply:
Since I am not sure it was clear:  I would suggest if we have a helper, it should be `_npy_import_numpy_multiarray_umath()`.

Reply:
@seberg How about `_npy_import_from_numpy_core(submodule_name)`?

In autogenerated headers `_multiarray_umath` is imported, but there's also `_internal` being imported from `_core` in `src/common/npy_ctypes.h`. This header is defined as a dependency for `libnpymath.a` so it's sourcecode ends up in downstream library (if I understand it correctly).

(A downstream lib could use `libnpymath.a` when compiling and then run with NumPy 1.x - this could cause import error on `numpy._core._internal`).

Reply:
> This header is defined as a dependency for libnpymath.a so it's 

Sorry, I am not seeing/understanding this.  Those should be private headers, and even *if* the function exposed there isn't used in `libmpymath.a`?  Where do you see them being a dependency?

I would be surprised to find any use of this in public API accidentally or not.  But, it wouldn't be the first surprise like this :).

Reply:
I thought `common/npy_ctypes.h` is included here https://github.com/numpy/numpy/blob/d676a1fe2d495f9d8a86103644bed141c2e69787/numpy/core/meson.build#L544, but it only defines where to search for header folders, right?
Then sorry for confusion!

Reply:
Ok, I finished refactoring - I ended up with a simpler version. 
(For some reason I couldn't import numpy in this initialization function (`PyImport_ImportModule("numpy")` returns null, maybe due to some circular import?)